app-admin/needrestart: should depend on >=sys-apps/sed-4.2.2

candrews opened the pull request:
needrestart installs and uses scripts that call sed with '-z' option:
  https://github.com/liske/needrestart/blob/1a3b68a38e0691ca82e894ef591b23af666cc14d/lib/notify.d.sh#L34
  https://github.com/liske/needrestart/blob/1c17aa96ed455c6f10b496fa6a5c17daf96558ac/ex/notify.d/400-notify-send#L34
This option was introduced in sed-4.2.2: http://article.gmane.org/gmane.comp.lang.smalltalk.gnu.general/7873
Thus needrestart must depend on >=sys-apps/sed-4.2.2.

https://bugs.gentoo.org/show_bug.cgi?id=588416

monsieurp commented:
@gentoo/proxy-maint

gktrk commented on app-admin/needrestart/needrestart-2.8.ebuild:
```diff
@@ -21,6 +21,7 @@ SLOT="0"
 LICENSE="GPL-2"
 
 RDEPEND="
+	>=sys-apps/sed-4.2.2
```
Adding a runtime dependency typically requires an ebuild revbump

gktrk commented:
The correct license for this package should be GPL-2+ per https://github.com/liske/needrestart/blob/master/perl/lib/NeedRestart.pm#L13

candrews commented:
sed is part of the system set so it was an implicit dependency - this change increases the min version required, and that doesn't require a rev bump. Right?

candrews commented:
monsieurp, I understand that a revbump is necessary - thank you for pointing that out. I've made the license change and did the revbump, so I think this PR is ready to be merged.

gktrk commented:
gettext is a build time dependency for this package:
```
make[1]: Entering directory '/var/tmp/portage/app-admin/needrestart-2.8-r1/work/needrestart-2.8/po/needrestart'
for po in de.po; do \
    lang=$(basename -s .po $po); \
    mkdir -p ../.build/$lang/LC_MESSAGES; \
    msgfmt -o ../.build/$lang/LC_MESSAGES/needrestart.mo $po; \
```

candrews commented:
@gktrk Thank you for noticing - I've added sys-devel/gettext as a build dependency and updated this PR.

gktrk commented:
I am not a Perl person but looks to me like we need `dev-perl/libintl-perl` as a runtime dependency per [Locale::TextDomain
](https://github.com/liske/needrestart/blob/master/INSTALL.md#perl)

candrews commented:
@gktrk I went through https://github.com/liske/needrestart/blob/master/INSTALL.md and checked the dependencies in the ebuilld against it:
* Added dev-perl/TermReadKey
* Removed dev-perl/Term-ProgressBar-Simple (and confirmed it truly isn't needed)

gktrk commented:
Merged in https://github.com/gentoo/gentoo/commit/89608b1c031f38cf19a2bc21212b854eb33d431d. Thanks!
